src/hooks/worlds/useWorldBySlug.ts

```diff
@@ -0,0 +1,68 @@
+import { useFirestore, useFirestoreCollectionData } from "reactfire";
+import Bugsnag from "@bugsnag/js";
+
+import { World } from "api/admin";
+
+import { worldConverter } from "utils/converters";
+import { WithId } from "utils/id";
+
+type UseWorldEditResult = {
+  world?: WithId<World>;
+  isLoaded: boolean;
+};
+
+/**
+ * Hook which will return the world when the slug is provided.
+ * The intention is to be used on the client side, when the world slug is provided in the url.
+ * @param worldSlug
+ * @returns
+ */
+export const useWorldBySlug: (worldSlug?: string) => UseWorldEditResult = (
+  worldSlug
+) => {
+  const firestore = useFirestore();
+
+  const worldsRef = firestore
+    .collection("worlds")
+    .where("slug", "==", worldSlug ?? "")
+    .withConverter(worldConverter);
+
+  const { data: worlds, status } = useFirestoreCollectionData<WithId<World>>(
+    worldsRef,
+    {
+      initialData: undefined,
+    }
+  );
+
+  const isWorldLoaded = status !== "loading";
+
+  if (!worlds?.[0]) {
+    Bugsnag.notify(
+      `World with the following slug: ${worldSlug} does not exist.`,
```

Comment:
I think we're going to see this a lot with bots crawling the site etc. Might be worth just throwing an error that eventually results in a 404

Reply:
My initial reaction was mostly about spamming our Bugsnag, but I kind of let it go to first check if that causes an issue, I mean, the solution would be a simple replacement of `Bugsnag.notify()` with `console.error()` just to log it somewhere.

I don't think relying on errors unwinding stack should be the normal behavior though, but an exception to handle in a nicer way.